src/clangsa/GCChecker.cpp

```diff
@@ -1106,11 +1107,19 @@ void GCChecker::checkDerivingExpr(const Expr *Result, const Expr *Parent,
     }
   }
   // This is the pointer
-  SValExplainer Ex(C.getASTContext());
-  auto ValLoc = C.getSVal(Result).getAs<Loc>();
-  if (!ValLoc) {
-    return;
+  auto ResultVal = C.getSVal(Result);
+  if (ResultVal.isUnknown()) {
+    if (!Result->getType()->isPointerType()) {
```

Comment:
Is this right? What about references? Also are you sure you want a symbol for the derived expr? Wouldn't it make more sense to have a symbol for the parent and then take a subregion for the derived expr?

Reply:
It's what you do everywhere else. And it's required for getting back something compatible with `ResultVal.getAs<Loc>()`.

Reply:
I don't know what this code does, since it's got almost no comments, so I just hammered at it until something passed. If you have a better way, we can do that.